Library/Homebrew/formula.rb

```diff
@@ -287,6 +287,15 @@ def version
     active_spec.version
   end
 
+  def update_head_version
+    return unless head?
+    head_resource = head.get_resource
+    if head_resource.downloader.is_a?(VCSDownloadStrategy) && \
+        head_resource.downloader.cached_location.exist?
+      head_resource.version.update_commit(head_resource.downloader.last_commit)
+    end
+  end
```

Comment:
I think this can be made a little bit more readable with additional early returns like this:

```ruby
def update_head_version
  return unless head?

  downloader = head.get_resource.downloader
  return unless downloader.is_a?(VCSDownloadStrategy)
  return unless downloader.cached_location.exist?

  head.get_resource.version.update_commit(downloader.last_commit)
end
```